src/models/geograph.py

```diff
@@ -43,20 +123,179 @@ def graph(self) -> nx.Graph:
+
+        Args:
+            vector_path (pathlib.Path): Path to a gpkg or shp file.
+            attributes (Optional[List[str]], optional): columns of the dataframe
+            that are added to nodes of graph as attributes. Defaults to None,
+            which will cause all attributes to be added.
+            load_slice: A slice object denoting the rows of the dataframe to
+            load. Defaults to None, meaning load all rows.
+
+        Raises:
+            ValueError: If `vector_path` is not a GPKG or Shapefile.
+
+        Returns:
+            STRtree: The Rtree object used to build the graph.
+        """
+        if vector_path.suffix not in (".shp", ".gpkg"):
+            raise ValueError("Argument `vector_path` should be a GPKG or Shapefile.")
+        # First try to load as GeoPackage, then as Shapefile.
+        if slice is not None:
+            dataframe = gpd.read_file(
+                vector_path, rows=load_slice, enabled_drivers=["GPKG", "ESRI Shapefile"]
+            )
+        else:
+            dataframe = gpd.read_file(
+                vector_path, enabled_drivers=["GPKG", "ESRI Shapefile"]
+            )
+        return self._dataframe_to_graph(
+            dataframe, attributes=attributes, tolerance=self.tolerance
+        )
+
+    def _load_raster(
+        self,
+        raster_path: Optional[Union[str, os.PathLike]],
+        save_path: pathlib.Path,
+        array: Optional[np.ndarray],
+        mask: Optional[np.ndarray],
+        transform: affine.Affine,
+        crs: Optional[str],
+    ) -> STRtree:
+        """
+        Load raster data, polygonise, then load graph and rtree.
+
+        The raster data should be in GeoTiff format.
+
+        Args:
+            raster_path (pathlib.Path): A path to a file of raster data in
+            GeoTiff format.
+            save_path (pathlib.Path, optional): A path to a file to save the
+            polygonised raster data in. A path to a GPKG file is recommended,
+            but Shapefiles also work.
+            array (np.ndarray, optional): 2D numpy array with the raster data
+            that can be passed as an alternative to `raster_path`.
+            mask (np.ndarray, optional): Boolean mask that can be applied over
+            the polygonisation. Defaults to None.
+            transform (affine.Affine, optional): Affine transformation to apply
+            when polygonising. Defaults to the identity transform.
+            crs (str, optional): Coordinate reference system to set on the
+            resulting dataframe. Defaults to None.
+
+        Raises:
+            ValueError: If `save_path` is not a GPKG or Shapefile, or if both
+            `raster_path` and `array` are None.
+
+        Returns:
+            STRtree: The Rtree object used to build the graph.
         """
-        Load networkx graph object from a pickle file.
+        if save_path.suffix not in (".shp", ".gpkg"):
```

Comment:
What happens if we want to work in memory only (i.e. not write to disk)? Say for example for a small test, or a brief interactive analysis in jupyter? Is that currently supported?

Reply:
It is not currently - I thought it best to have the default behaviour to be to save files (though if you do it repeatedly it just overwrites the same file). Should I add in an extra boolean argument in the init that turns off all saving?

Reply:
Update: I turned off saving by default so if you don't pass any save paths then it doesn't do anything.